source/options.html

```diff
@@ -23,6 +23,13 @@ <h2 class="only-firefox">Options</h2>
 		</label>
 	</p>
 
+	<p>
+		<label>
+			<strong>Personal token</strong>, <a href="https://github.com/settings/tokens/new?description=Refined%20GitHub">found here</a> (for high rate API calls and private repos)<br>
+			<input name="personalToken" spellcheck="false" autocomplete="off" size="40" maxlength="40" pattern="[a-z0-9]{40}" placeholder="Must be 40 characters in length"></input>
```

Comment:
The token is hex, so the pattern should be `[0-9a-f]{40}`

Reply:
Also we don't need the placeholder, we already give a link to the source, which should be copy-pasted

Reply:
I considered the hex pattern, but thought it might be too brittle in the event of changes by GitHub. I'll add it though per your request.

The placeholder is actually present so that we can use the `not(:placeholder-shown)` selector when styling the `input`, so it's not green by default.

Other options are:

1. Remove the green style (only show when it's invalid)
2. Use `placeholder=""` instead (which is a bit of a hack)

Reply:
In my other extension I used the `required` attribute which eliminated the issue (because the key was actually required), but I'm not sure if it brings any side effects here. Perhaps we can try that